pom.xml

```diff
@@ -56,8 +58,109 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>ssh-slaves</artifactId>
-            <version>1.25</version>
+            <version>1.27</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>structs</artifactId>
+            <version>1.17</version>
             <scope>test</scope>
         </dependency>
     </dependencies>
+    <profiles>
+        <profile>
+            <id>docker-generate-binaries</id>
+            <activation>
+                <os>
+                    <family>unix</family>
```

Comment:
This seems unwise. It means that any random developer using Windows who runs `mvn package` will get a `durable-task.hpi` which is in fact unusable, with no warning. (Conversely, people with Windows subsystem for Linux will not be able to do builds, perhaps.) That is one reason I suggested the two-stage `Jenkinsfile` approach.

Reply:
removed. will work on implementing two-stage jenkinsfile approach